File: my_vpn/app/src/main/java/com/myvpn/simple/LibXivpn.java
package com.myvpn.simple;

public class LibXivpn {
    
    static {
        System.loadLibrary("xivpn");
    }
    
    public static native String xivpn_version();
    
    public static native String xivpn_start(String config, int socksPort, int fd, String logFile, String asset, SocketProtect protect);
    
    public static native void xivpn_stop();
    
    public static native void xivpn_init();
}

File: kt/app/src/main/kotlin/cn/gov/xivpn2/LibXivpn.kt
package cn.gov.xivpn2

import cn.gov.xivpn2.service.SocketProtect

object LibXivpn {
    @JvmStatic
    external fun xivpn_version(): String

    @JvmStatic
    external fun xivpn_start(
        config: String,
        socksPort: Int,
        fd: Int,
        logFile: String,
        asset: String,
        protect: SocketProtect
    ): String

    @JvmStatic
    external fun xivpn_stop()

    @JvmStatic
    external fun xivpn_init()
}
